openc3/lib/openc3/models/python_package_model.rb

```diff
@@ -71,16 +74,23 @@ def self.install(name_or_path, scope:)
       package_filename = File.basename(package_file_path)
       begin
         pypi_url = get_setting('pypi_url', scope: scope)
-      rescue
-        # If Redis isn't running try the ENV, then simply pypi.org/simple
-        pypi_url = ENV['PYPI_URL']
         if pypi_url
           pypi_url += '/simple'
         end
-        pypi_url ||= 'https://pypi.org/simple'
+      rescue => e
+        Logger.error("Failed to retrieve pypi_url: #{e.formatted}")
+      ensure
+        if pypi_url.nil?
+          # If Redis isn't running try the ENV, then simply pypi.org/simple
+          pypi_url = ENV['PYPI_URL']
+          if pypi_url
+            pypi_url += '/simple'
+          end
+          pypi_url ||= 'https://pypi.org/simple'
+        end
       end
       Logger.info "Installing python package: #{name_or_path}"
-      result = OpenC3::ProcessManager.instance.spawn(["pip", "install", "--user", "-i", pypi_url, package_file_path], "package_install", package_filename, Time.now + 3600.0, scope: scope)
+      result = OpenC3::ProcessManager.instance.spawn(["/openc3/bin/pipinstall", "--user", "--no-warn-script-location", "-i", pypi_url, package_file_path], "package_install", package_filename, Time.now + 3600.0, scope: scope)
```

Comment:
Looks like we were using this in the plugin_model.rb but not here. Is this the real fix?

Reply:
This adds a retry if the remote install fails, that try to just do local.